guides/articles/self-host-hoppscotch-on-your-own-servers.mdx

```diff
@@ -0,0 +1,283 @@
+    > 
+    > 
+    > ```jsx
+    > ip addr show docker0
+    > ```
+    > 
+    > Look for the inet address associated with the docker0 interface. It’s typically in the **172.17.0.0/16** range but may vary based on your Docker network configuration. And If you’re using **Docker Desktop**, you can use `host.docker.internal` instead of `localhost`.
+    > 
+    
+    ```bash
+    docker run --name='mailcatcher' -d \\
+      --publish=<Docker_bridge_IP or host.docker.internal>:1080:1080 \\
+      --publish=<Docker_bridge_IP or host.docker.internal>:1025:1025 \\
+      dockage/mailcatcher:0.9.0
+    ```
+    
+    Visit `http://<Docker_bridge_IP or host.docker.internal>:1080` to access the Mailcatcher web interface and view email communications.
+    
+    Further, configure the below environment variables in your `.env` file:
+    
+    ```jsx
+    MAILER_SMTP_ENABLE=true
+    MAILER_USE_CUSTOM_CONFIGS=false
+    MAILER_ADDRESS_FROM=from@example.com
+    MAILER_SMTP_URL=smtp://<Docker_bridge_IP or host.docker.internal>:1025
+    ```
+    
+
+#### **2.2 Custom Mailer Configuration**
+
+For advanced email delivery needs, such as for production environments, you can configure a custom email service by setting `MAILER_USE_CUSTOM_CONFIGS=true`. You can choose from services like [SendGrid](https://sendgrid.com/), [Amazon SES](https://aws.amazon.com/ses/), or your own SMTP server. Once you’ve set up your chosen service, update your `.env` file with the following details:
+
+```jsx
+MAILER_SMTP_HOST=smtp.domain.com
+MAILER_SMTP_PORT=587
+MAILER_SMTP_SECURE=true
+MAILER_SMTP_USER=user@domain.com
+MAILER_SMTP_PASSWORD=pass
+MAILER_TLS_REJECT_UNAUTHORIZED=true
+```
+
+### 3. OAuth Configuration
+
+To access the admin dashboard, you’ll need to configure an OAuth provider. In the Community Edition, Hoppscotch supports:
+
+1. Google
+2. GitHub
+3. Microsoft
+
+In the Enterprise Edition, support also includes **SAML SSO, OpenID Connect, and GitHub Enterprise**.
+
+Here’s a quick guide to registering an OAuth application with GitHub:
+
+1. Click your profile photo in the upper-right corner and select **Settings**.
+2. In the left sidebar, scroll down and click **Developer Settings**.
+3. Click **OAuth Apps** in the sidebar.
+4. Click **New OAuth App**.
+5. Enter the required information and specify the callback URL as indicated in your configuration.
+6. After registering the application, copy the Client ID and Client Secret, and add them to your environment file.
+
+Similarly, you can follow the specific setup instructions for other OAuth providers to complete your configuration.
+
+### 4. Subpath Access
+
+Subpath access allows you to host multiple services under a single domain by assigning each service a specific subpath.
+
+To enable subpath-based access, set `ENABLE_SUBPATH_BASED_ACCESS=true` (the **default** setting is **false**).
+
+When enabled, here's how it affects your setup:
+
```

Comment:
Lets just replace this whole section and just say if subpath is enabled you can access the 3 services on the same domain but different routes and if its not enabled you access them on different ports.

Reply:
Done